src/vmm/src/persist.rs

```diff
@@ -303,6 +307,34 @@ pub fn validate_x86_64_cpu_vendor(
     Ok(())
 }
 
+/// Performs sanitychecks against the state file and returns specific errors.
+pub fn snapshot_state_sanity_check(
+    microvm_state: &MicrovmState,
+) -> std::result::Result<(), LoadSnapshotError> {
```

Comment:
nit: all functions in this file return `LoadSnapshotError` result, and now there's enough of them to make it worth using a shorthand
```rust
type Result<T> = std::result::Result<T, LoadSnapshotError>;
```